hrjobcontract/CRM/Hrjobcontract/Dedupe/Merger.php

```diff
@@ -601,27 +624,87 @@ static function batchMerge($rgid, $gid = NULL, $mode = 'safe', $autoFlip = TRUE,
     return CRM_Hrjobcontract_Dedupe_Merger::merge($dupePairs, $cacheParams, $mode, $autoFlip, $redirectForPerformance);
   }
 
+  public static function updateMergeStats($cacheKeyString, $result = array()) {
+    // gather latest stats
+    $merged  = count($result['merged']);
+    $skipped = count($result['skipped']);
+
+    if ($merged <= 0 && $skipped <= 0) {
+      return;
+    }
+
+    // get previous stats
+    $previousStats = CRM_Core_BAO_PrevNextCache::retrieve("{$cacheKeyString}_stats");
+    if (!empty($previousStats)) {
+      if ($previousStats[0]['merged']) {
+        $merged = $merged + $previousStats[0]['merged'];
+      }
+      if ($previousStats[0]['skipped']) {
+        $skipped = $skipped + $previousStats[0]['skipped'];
+      }
+    }
```

Comment:
What about extracting this to a new method? It would make the code easier to read (you wouldn't need the "get previous stat" comment) and make this method smaller.